### PR TITLE
feat: Explicitly mark exported API

### PR DESCRIPTION
**Pull request by LecrisUT**
Windows systems are quite weird where you have to define both when you import and export stuff. Previously in the CI we got around it by building as static, but that is not appropriate as we've seen with the C# interface discussion [^1].

@husakm can you try this one out?

[^1]: https://github.com/spglib/spglib/discussions/399#discussioncomment-8207195

### CODE REVIEW

**Comment by atztogo** [comment]
@LecrisUT, could you tell me how the macros in `base.h` work or the aim unless it is written somewhere?

**Comment by LecrisUT** [comment]
Basically the default symbol visibility is changed to hidden, but that means the unit tests would not be able to find the symbols, even though it has access to the internal headers. So, as a workaround, we patch the logic to export those symbols when the test-suite is being built with unit tests.

The other approaches I've seen are:
- build the static library together with the shared one. This has a lot of overhead downstream because they would have to run the configure-build-test twice
- make maintenance hatch exposed interfaces, i.e. a wrapper that exposes the api and with test nomenclature. Since the headers are not public, the current approach effectively makes them maintenance hatch interface, so no need to be explicit

File: src/base.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#ifndef SPGLIB_BASE_H
+#define SPGLIB_BASE_H
+
+#include "spglib.h"
+
+#ifdef SPG_TESTING
+#define SPG_API_TEST SPG_API
```
Comment (atztogo) [inline src/base.h:7]: Where is `SPG_API` used?

**Review by atztogo — APPROVED** [review]

File: src/base.h (hunk 1 above)
Comment (LecrisUT) [reply src/base.h:7]: It's defined in `spglib.h` and it's used to export/import funtions to/from dll. On unix it only does export

**Comment by lan496** [comment]
Can you leave a document on `SPG_API_TEST` in Contributing.md or somewhere appropriate? Small note on why this macro is needed will be enough.

**Comment by LecrisUT** [comment]
Ok, I've added a note about `SPG_API_TEST` in `test/README.md`. I think we should move it when we refactor the documentation structure.

**Review by lan496 — APPROVED** [review]
Thanks. The note `Exposing API for unit tests` looks concise and simple for contributors.

**Pull request merged**